### DISCUSSION TITLE
Param encoding of square brackets

### DISCUSSION THREAD

**Discussion by whre** [axios/axios#5607] [Q&A]
Hi, axios by default seems to not encode square brackets contained within string values, that are supplied as query params.
I saw another discussion describing that square brackets are used to represent multi-value-params in the query string.
But still it seems to be required that "[]" are encoded as "%5B%5D" then https://stackoverflow.com/questions/40568/are-square-brackets-permitted-in-urls (else servers like in spring boot deny those characters as invalid).
Is there some setting, to have these characters %-encoded like all the other special query characters?

**Comment by jg123** [comment]
Subscribe here: https://github.com/axios/axios/pull/5715
